cmd/sourced/release/release.go

```diff
@@ -0,0 +1,50 @@
+// Package release deals with versioning and releases
+package release
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/src-d/sourced-ce/cmd/sourced/dir"
+
+	"github.com/blang/semver"
+	"github.com/google/go-github/v25/github"
+	"github.com/gregjones/httpcache"
+	"github.com/gregjones/httpcache/diskcache"
+)
+
+// FindUpdates calls the GitHub API to check the latest release tag. It returns
+// true if the latest stable release is newer than the current tag, and also
+// that latest tag name.
+func FindUpdates(current string) (update bool, latest string, err error) {
+	currentV, err := semver.ParseTolerant(current)
+	if err != nil {
+		return false, "", err
+	}
+
+	diskcachePath := filepath.Join(dir.TmpPath(), "httpcache")
+	err = os.MkdirAll(diskcachePath, os.ModePerm)
+	if err != nil {
+		return false, "", err
+	}
+
+	cache := diskcache.New(diskcachePath)
+	client := github.NewClient(&http.Client{Transport: httpcache.NewTransport(cache)})
```

Comment:
default caching time is 60 seconds. Maybe we should check it even less often?

<img width="640" alt="Screenshot 2019-06-04 at 14 41 26" src="https://user-images.githubusercontent.com/406916/58879903-ff8c0300-86d6-11e9-8113-6f2346793642.png">

Reply:
The github.com/gregjones/httpcache lib makes use of `If-None-Match` and `If-Modified-Since`, so it takes care of https://developer.github.com/v3/#conditional-requests and it doesn't count for the quota. If that's what you are concerned about.

Reply:
thanks. true. cache-control is only the very first step. if it expired it uses etag